helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixManager.java

```diff
@@ -697,10 +699,19 @@ void createClient() throws Exception {
     int retryCount = 0;
     while (retryCount < 3) {
       try {
-        // TODO: synchronize this block and wait for the new non-zero session ID updated.
-        _zkclient.waitUntilConnected(_connectionInitTimeout, TimeUnit.MILLISECONDS);
+        // TODO: get session id from waitUntilConnected to avoid race condition
```

Comment:
This is OK. Just a reminder that we need to finish this TODO until the bug is finished.

Reply:
Will submit a new PR for this.